client/src/lib.rs

```diff
@@ -153,31 +150,76 @@ impl TurnkeyClient {
         }
     }
 
-    pub fn current_timestamp(&self) -> String {
+    /// Returns the current timestamp as a u128.
+    pub fn current_timestamp(&self) -> u128 {
         let now = SystemTime::now();
 
         now.duration_since(UNIX_EPOCH)
             .expect("Time went backwards")
             .as_millis()
-            .to_string()
+    }
+
+    /// Processes a `Request` (at `path`) by:
+    /// * Serializing the request to JSON
+    /// * Signing the request with the client's API key
+    /// * POSTing the POST body and the associated stamp to the Turnkey API
+    ///
+    /// This function is generic and can handle POSTing queries or activities.
+    pub async fn process_request<Request, Response>(
+        &self,
+        request: &Request,
+        path: String,
+    ) -> Result<Response, TurnkeyClientError>
+    where
+        Request: Serialize,
+        Response: DeserializeOwned,
+    {
+        let url = format!("{}{}", self.base_url, path);
+        let post_body = serde_json::to_string(&request)?;
+        let stamp = self.api_key.stamp(post_body.clone())?;
+        let res = self
+            .http
+            .post(url)
+            .header("X-Stamp", stamp)
+            .body(post_body)
+            .send()
+            .await?;
+
+        let status = res.status();
+        let text = res.text().await?;
+
+        if !status.is_success() {
```

Comment:
Shouldn't we look at the Content-Type to decide if it's json or not?
Just because something parses as valid json doesn't meant it's *meant* be json

Reply:
> Our API should all be JSON. So if the content-type isn't JSON I can safely error out (with a proper error type).

It's also entirely possible that our reverse proxy (traefik) in certain situations responds with json document error messages of its own (e.g. if a service is internally unreachable)

Reply:
Done, I think the handling is a lot more robust now! Also, added a test to make sure we fail with the right error.